### PR TITLE
feat: showing required flag

### PR DESCRIPTION
**Pull request by c-pius**
**Description**

Changes proposed in this pull request:

- showing a "required" flag for required properties (as also done in the [playground](https://playground.asyncapi.io/))

**Related issue(s)**

none

### CODE REVIEW

**Comment by c-pius** [comment]
required-flag

the style is similar to how it is done in the [playground](https://playground.asyncapi.io/). Unfortunately I couldn't find out how to properly indent the "required" text if it is part of the tree. I assume it has something to do with the TreeSpace but couldn't really find out here https://github.com/asyncapi/asyncapi-react/blob/daf9f13440a8218713c8f678fae6b676c834c0cc/library/src/containers/Schemas/SchemaProperties.tsx#L125

**Comment by c-pius** [comment]
the [playground](https://playground.asyncapi.io/) also shows a required flag for channel parameters. I assume this could be done as well by passing a required=true flag to the SchemaComponent and chaining it through:

https://github.com/asyncapi/asyncapi-react/blob/daf9f13440a8218713c8f678fae6b676c834c0cc/library/src/containers/Channels/Parameter.tsx#L26

https://github.com/asyncapi/asyncapi-react/blob/daf9f13440a8218713c8f678fae6b676c834c0cc/library/src/containers/Schemas/Schema.tsx#L38

should I do that as well?

**Comment by magicmatatjahu** [comment]
@c-pius Hello! I don't have any problems with changed code. About indentations: exactly, you can use `TreeSpace` component like [here](https://github.com/asyncapi/asyncapi-react/blob/3d2f0284239f2e7773c28d0c8e2d0869d8c323bc/library/src/containers/Schemas/SchemaProperties.tsx#L131), so using `treeSpace` prop you add indentations in `for loop` :) About parameter: if you want, feel free to do it :)

**Comment by c-pius** [comment]
@magicmatatjahu great. So parameters are now always shown as required. Indentation also works (it is not 100% aligned as you can see from the screenshot but I would consider it okay).

Screenshot 2021-01-14 at 10 26 34

**Comment by magicmatatjahu** [comment]
@c-pius Thanks! Looks great :) If you want, you can manipulate with left padding to adjust the required to top text.

File: library/src/containers/Channels/Parameter.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const Parameter: FunctionComponent<Props> = ({
         schema={schema}
         hideTitle={true}
         description={description}
+        required={true} // parameters are always required
```
Comment (fmvilas) [inline library/src/containers/Channels/Parameter.tsx:32]: Good catch 👍

**Review by fmvilas — APPROVED** [review]
Thanks, @c-pius!

**Comment by c-pius** [comment]
@magicmatatjahu
> If you want, you can manipulate with left padding to adjust the required to top text.

Couldn't really get it to work with padding-left so I inserted this one. Not sure if this is dirty? 

```
          if (el.treeSpace) {
            for (let i = 0; i < el.treeSpace; i++) {
              treeSpaces.push(<TreeSpace key={i} />);
            }
            // adding 25px span to account for the missing TreeLeaf
            treeSpaces.push(
              <span style={{ display: 'inline-block', width: '25px' }} />,
            );
          }
```

**Comment by magicmatatjahu** [comment]
@c-pius Ok, you can leave it, yes it's dirty 😄 We don't want to make inline css, because component has a possibility to override css, and we shouldn't override this feature.

**Comment by c-pius** [comment]
alright, reverted the dirty stuff 😄

**Comment by c-pius** [comment]
> EDIT: First we want to merge this PR, before your one #218

sure, nor hurry.

**Comment by derberg** [comment]
@c-pius please update your pr with latest master

@magicmatatjahu I merged, whenever you are ready with this one, approve and merge and let's see how this new CI beast works 😅

**Comment by c-pius** [comment]
@derberg ready

**Review by magicmatatjahu — APPROVED** [review]

**Comment by c-pius** [comment]
🙌🏻  thanks!

**Comment by derberg** [comment]
@all-contributors please add @c-pius for code, bug

**Pull request merged**